devcontainer-setup.md

```diff
@@ -0,0 +1,226 @@
+
+**Sample repository**:
+
+https://github.com/jetbrains-junie/kotlin-spring-chat
+https://github.com/jetbrains-junie/kotlin-spring-chat/blob/main/.devcontainer/devcontainer.json
+https://github.com/jetbrains-junie/kotlin-spring-chat/blob/main/Dockerfile
+
+---
+
+## Scenario 3: `devcontainer.json` + `docker-compose.yml` (Multi-Container with Pre-built Images)
+
+Use this when your application requires multiple services (e.g., app + database) and these services can use pre-built images.
+
+**When to use:**
+*   You need to orchestrate multiple services (e.g., web app, database, cache).
+*   The services can be defined using existing images from a registry.
+*   You want to manage service configurations (ports, env vars, networks, volumes) in `docker-compose.yml`.
+
+**File Structure:**
+```
+your-project/
+├── .devcontainer/
+│   ├── devcontainer.json
+│   └── docker-compose.yml    // Defines your services
+└── (your project files)
+```
+
+**`docker-compose.yml` Example:**
+```yaml
+services:
+  devcontainer_app:
+    image: 'mcr.microsoft.com/devcontainers/java:1-17'
+    volumes:
+      # mount project's parent folder as /workspace,
+      # so the project itself will be available at /workspace/<project-name>
+      # local path is relative to docker-compose.yml
+      - ../../:/workspace/:cached
+    command: 'sleep infinity'
+    # network with service_db
+    network_mode: 'service:postgres_db'
+
+  postgres_db:
+    image: 'postgres:latest'
+    environment:
+      - 'POSTGRES_DB=mydatabase'
+      - 'POSTGRES_PASSWORD=secret'
+      - 'POSTGRES_USER=myuser'
+    ports:
+      - '5432'
+```
+
+**`.devcontainer/devcontainer.json` Example:**
+```json
+{
+  "name": "Java",
+  // relative path to docker-compose.yaml which will be used to launch the devcontainer
+  "dockerComposeFile": "docker-compose.yaml",
+  // particular service from "dockerComposeFile" which will be used as a primary container.
+  // sources will be mounted into that container at "workspaceFolder"
+  "service": "devcontainer_app",
+  "workspaceFolder": "/workspace/${localWorkspaceFolderBasename}",
+  "features": {
+    "ghcr.io/devcontainers/features/java:1": {
+      "version": "none",
+      "installMaven": "true",
+      "mavenVersion": "3.8.6"
+    }
+  }
+}
+```
+**Explanation:**
+*   `dockerComposeFile` points to your `docker-compose.yml`.
+*   `service` specifies which service defined in `docker-compose.yml` devcontainer should use (in this case, `devcontainer_app`).
+*   `workspaceFolder` is the path *inside the specified service's container*.
+*   Docker Compose handles starting all defined services and their dependencies.
+
+**Note:**
+* similar to the Dockerfile, you need to explicit mount `workspaceFolder` to `/workspace/${localWorkspaceFolderBasename}`.
+  Since, variables are not available inside `docker-compose.yml` we mount project's parent directory to `/workspace/`
+
+**Sample repository**:
+
+https://github.com/jetbrains-junie/workout-logger-demo
```

Comment:
(the same)

- Check the formatting.(like example with links, devcontainer example link, docker-compose example link)
- Rename repositories and ensure all repositories are public.

Additionally, it would be great to have an example where we have JVM repositories, and Junie fixes something and runs the tests.